modules/core/arrow-core-data/src/test/kotlin/arrow/core/OptionTest.kt

```diff
@@ -137,6 +139,27 @@ class OptionTest : UnitSpec() {
       None or None shouldBe None
 
     }
+
+    "fx can turn effects into pure kinded values" {
+      suspend fun sideEffect(): Int =
+        1
+      fx {
+        val (result) = f { sideEffect() }
```

Comment:
I got private feedback from some people I asked from the FP community in Scala, Haskell and Kotlin.
`f` turns a suspended function that denotes a side effect into a `Kind<F, A>` and it's available in all `fx` blocks. It then lifts the side effect to the monad in question, in this case `Option`. When it injests the side effect it tries to lift `A` resulting from the effect into the monadic context.
In option it will throw an exception if it failed, in other data types capable of monad error captures the exception and raises it to F and in others that are suspend capable like MonadDefer, Async or Concurrent it does the appropriate defer op to suspend side effects.
```kotlin
fun <A> f(fa: suspend () -> A): Kind<F, A>
```
We may rename it or alias it to something more meaningful. currently the Fx primitives are:
- `fx`. A monadic binding block 
- effect. Injest an external side effect and bind
- f. Injest a side effect and not bind but instead get the `Kind<F, A>` value
- unsafe. explicitly perform side effects at the edge of the world